src/apps/external_executor/external_executor.cpp

```diff
@@ -161,6 +162,52 @@ namespace externalexecutor
         .install();
     }
 
+    struct AsyncMsg
+    {
+      ccf::grpc::StreamPtr<externalexecutor::protobuf::KVKeyValue> stream;
+
+      AsyncMsg(
+        const ccf::grpc::StreamPtr<externalexecutor::protobuf::KVKeyValue>&
+          stream_) :
+        stream(stream_)
+      {}
+    };
+
+    static void send_stream_payload(
+      const ccf::grpc::StreamPtr<externalexecutor::protobuf::KVKeyValue>&
+        stream,
+      bool close_stream)
+    {
+      externalexecutor::protobuf::KVKeyValue kv;
+      kv.set_key("my_key");
+      kv.set_value(fmt::format("my_value: {}", close_stream));
+
+      stream->stream_msg(kv, close_stream);
+    }
+
+    static void async_send_stream_data(
```

Comment:
I think we should add a pub/sub endpoint (ie - one streaming endpoint which gets another message contains every time `/foo` is called) as a sample, in this PR. The underlying implementation is also tested by this endpoint, but that would prove out the `Stream` ownership model/reachability a little more (this sample still triggers everything on the `Stream` from a single user call - albeit transitively via the delayed thread message - and we want to confirm we can also write responses triggered from other endpoints, or hooks, or indexes, or...)

Reply:
I've added a `/Pub` and `/Sub` endpoints. I'll leave the `/Stream` endpoint in this PR until just before this PR is merged just to demonstrate multiple examples.